slang.h

```diff
@@ -2670,6 +2670,14 @@ namespace slang
             SlangPassThrough passThrough,
             const char* preludeText) = 0;
 
+            /** Get the 'prelude' for generated code for a 'downstream compiler'.
+            @param passThrough The downstream compiler for generated code that will have the prelude applied to it. 
+            @param outPrelude  On exit holds a blob that holds the string of the prelude.
+            */
+        virtual SLANG_NO_THROW void SLANG_MCALL getDownstreamCompilerPrelude(
```

Comment:
Irrespective on the relative merits of our prelude mechanism, it could be argued that our API for state that can be set, should also have a mechanism to get the state. 
In terms of prelude mechanism (or something like it) is likely going necessary for C/C++ targets, because C/C++ are so much more nuanced than other targets we have. 
The use that led to this inclusion in the code here is perhaps not particularly exemplary - and more a question of convenience.